src/fillalgebra.jl

```diff
@@ -261,6 +261,33 @@ end
 -(a::AbstractRange{T}, b::Zeros{V, 1}) where {T, V} = a + b
 
 
+function +(a::Zeros{T, 2}, b::UniformScaling{V}) where {T, V<:Number}
+    n = LinearAlgebra.checksquare(a)
+    return Diagonal(Fill{promote_type(T,V)}(b.λ, n))
+end
+function +(a::RectOrDiagonal{T, <:AbstractFill{T, 1}}, b::UniformScaling) where {T}
+    LinearAlgebra.checksquare(a)
+    return Diagonal(a.diag .+ b.λ)
+end
+
+function -(a::UniformScaling{T}, b::Zeros{V, 2}) where {T <: Number, V}
+    n = LinearAlgebra.checksquare(b)
+    return Diagonal(Fill{promote_type(T,V)}(a.λ, n))
+end
```

Comment:
In principle, this definition could be removed since the default definition in src/FillArrays.jl would be `-(a::UniformScaling, b::Zeros) = a + (-b)` which by default (in LinearAlgebra) ends up calling `(-b) + a` which is covered by the definition for `+` above. However, adding this definition seemed more direct, clearer, and simpler to reason about.

Reply:
With my suggestion above the special case for `Zeros` is not needed.

Reply:
I removed the definition for `Zeros`.